### ISSUE TITLE
Subtitle Support.

### ISSUE THREAD

**Issue by Mehmeed** [OstlerDev/PopcornTV#18]
What i think is that the only thing missing is subtitle support. it would be amazing! thanks

**Comment by OstlerDev** [comment]
In order to get subtitle support I need to write a bridge and methods to re-encode the video to include the Subtitles, This is planned for future implementation but other things need to be done first.

**Comment by Mehmeed** [comment]
oh ok thanks..

**Comment by OstlerDev** [comment]
I am going to leave this open until I implement the feature.

**Comment by pitercio** [comment]
Have you seen how subtitles are managed on the PlexConnect forum?

https://forums.plex.tv/index.php/topic/95057-official-new-subtitle-renderer-feedback-thread/

I think that properly formatted .SRT files can be shown on the Apple TV side without needing any transcoding. Perhaps the python script that deals with the subs is clearer by itself than the forum thread:

https://github.com/iBaa/PlexConnect/blob/master/Subtitle.py

Have a look :)

Regards

**Comment by OstlerDev** [comment]
I had actually not taken a look at that file. I had a suspicion that I could send SRT files along with the XML and the ATV would parse it, but I had not tested it at all. I will give that a more thourough lookover tomorrow when I am of a more sane mind. I have been messing with FFMPEG to try to be able to stream AVI files all of tonight.

**Comment by pitercio** [comment]
By the way, FANTASTIC EFFORT!! I'm looking forward to enjoying a combined PlexConnect/PopcornTV solution :)

**Comment by OstlerDev** [comment]
Thanks! I will be adding in support for both at some point in the future (you would need to run the PlexConnect server on a different computer most likely) however it comes down to all the media you would want (other then home movies and such) would be on PopcornTV anyways.

It is something I plan on adding down the road however.

**Comment by pitercio** [comment]
Hmmm... most of my media are spanish-dubbed, not currently available on PopcornTime. That's why I'm so interested in getting subtitles working :)

**Comment by OstlerDev** [comment]
A quick little update for people interested: https://www.youtube.com/watch?v=LqqyQ3zHUuM

I have gotten the basic implementation working as well as a basic subtitles API working. I am working on the second half right now and hope to have it all working by the end of the night!

**Comment by pitercio** [comment]
Cool!! Looking forward to trying it!!! :)

**Comment by OstlerDev** [comment]
If you could both do some testing and let me know how the feature works that would be great @Mehmeed @pitercio

Enjoy!

**Comment by domidyon** [comment]
Hi! First of all, what a fantastic app you made! I have one problem though, I get through the set up and all fine, but I can't finish the last step.. Installing the profile on the ATV. I just type in http://trailers.apple.com/trailers.cer but it says that it can't be added and that i should check the URL. Error code 109.. What do I do now?

**Comment by OstlerDev** [comment]
This is the wrong place to report this issue. You are getting error 109 because you did not change the DNS server on the Apple TV. Follow the installation instructions and open a new issue if the error persists.

**Comment by tomillo** [comment]
I'm using your app, but the list of subtitles are always empty. It's a bug on my machine or it's a bug on the app?
